common/ast/src/parser/statement.rs

```diff
@@ -759,6 +784,129 @@ pub fn column_def(i: Input) -> IResult<ColumnDefinition> {
     )(i)
 }
 
+//{ ROLE <role_name> | schemaObjectPrivileges | ALL [ PRIVILEGES ] ON <privileges_level> }
+pub fn grant_source(i: Input) -> IResult<GrantSource> {
+    let role = map(
+        rule! {
+            ROLE ~ #literal_string
+        },
+        |(_, role_name)| GrantSource::Role { role: role_name },
+    );
+    let privs = map(
+        rule! {
+            #comma_separated_list1(priv_type) ~ ON ~ #grant_level
+        },
+        |(privs, _, level)| GrantSource::Privs {
+            privileges: privs,
+            level,
+        },
+    );
+    let all = map(
+        rule! { ALL ~ PRIVILEGES? ~ ON ~ #grant_level },
+        |(_, _, _, level)| GrantSource::ALL { level },
+    );
+
+    rule!(
+        #role
+        | #privs
+        | #all
+    )(i)
```

Comment:
You can add the grammer comments to the hint, and then we can also remove L787:

```suggestion
    rule!(
        #role : "ROLE <role_name>"
        | #privs : "<privileges>, ..."
        | #all : "ALL [ PRIVILEGES ] ON <privileges_level>"
    )(i)
```

Same to the others.